### PR TITLE
Prevent getCollection from breaking in vitest

### PR DESCRIPTION
**Pull request by matthewp**
## Changes

- Threads through Vite's `mode` which is mutated by vitest.
- Fixes https://github.com/withastro/astro/issues/10822

## Testing

- https://stackblitz.com/edit/astro-vitest-image-content-collections-atavo4?file=src%2Futils%2Fcollections.test.ts,vitest.config.ts `npm run test`

## Docs

N/A, bug fix

### CODE REVIEW

**Comment by matthewp** [comment]
!preview assets-vitest

File: packages/astro/src/assets/vite-plugin-assets.ts [hunk 1]
```diff
@@ -194,6 +195,9 @@ export default function assets({
 		{
 			name: 'astro:assets:esm',
 			enforce: 'pre',
+			config(_, env) {
+				viteMode = env.mode;
+			},
```
[the remark this answers is not in the document]
Comment (matthewp) [reply packages/astro/src/assets/vite-plugin-assets.ts:200]: Yeah I can switch to that.

[the remark this answers is not in the document]
Comment (matthewp) [reply packages/astro/src/assets/vite-plugin-assets.ts:200]: This is used in the external API `emitESMImage()` so we can't rely on `command` being passed in a backwards-compatible way. But I can pass this through in internal code and use `watchMode` only if command is not provided.

**Comment by matthewp** [comment]
!preview assets-vitest

**Review by ematipico — APPROVED** [review]

**Review by bluwy — COMMENTED** [review]
Late to review, but the change looks great to me 👍

**Pull request merged**